pet-store-vanilla-js/app.js

```diff
@@ -105,26 +89,39 @@ function createViewEditButton(petId) {
   const viewEditButton = document.createElement('button');
   viewEditButton.textContent = 'View / Edit';
   viewEditButton.classList.add('btn', 'btn-warning');
-
-  viewEditButton.addEventListener('click', async () => {
-    await configureFormEditModal(petId);
-  });
+  viewEditButton.dataset.petId = petId;
+  viewEditButton.dataset.type = "edit";
 
   return viewEditButton;
 }
 
-function createDeleteButton(pet) {
+function createDeleteButton(petId) {
   const deleteButton = document.createElement('button');
   deleteButton.textContent = 'Delete';
   deleteButton.classList.add('btn', 'btn-danger');
-
-  deleteButton.addEventListener('click', async () => {
-    await showDeleteModal(pet);
-  });
+  deleteButton.dataset.petId = petId;
+  deleteButton.dataset.type = "delete";
 
   return deleteButton;
 }
 
+mainPageElements.tableBody.addEventListener('click', async (e) => {
+  if (e.target.tagName !== 'BUTTON') {
+    return;
+  }
+
+  const dataSet = e.target.dataset;
+  const petId = Number(dataSet.petId);
+  
+  const buttonType = dataSet.type;
+  if(buttonType === 'edit') {
+    await configureFormEditModal(petId);
+  } else if (buttonType === 'delete') {
+    const pet = allPets.find((x) => x.petId === petId);
```

Comment:
I don't like that you are looping through all of the pets here. Maybe keep a map that same way you do for the pet kinds?